pkg/serverless/serverless.go

```diff
@@ -158,7 +158,8 @@ func WaitForNextInvocation(stopCh chan struct{}, daemon *daemon.Daemon, id regis
 }
 
 func callInvocationHandler(daemon *daemon.Daemon, arn string, deadlineMs int64, safetyBufferTimeout time.Duration, requestID string, invocationHandler InvocationHandler) {
-	userCPUTimeMsOffset, systemCPUTimeMsOffset, cpuOffsetErr := proc.GetCPUData("/proc/stat")
+	userCPUTimeMsOffset, systemCPUTimeMsOffset, cpuOffsetErr := proc.GetCPUData(proc.ProcStatPath)
+	networkOffsetData, networkOffsetErr := proc.GetNetworkData(proc.ProcNetDevPath)
```

Comment:
Just something I'm chewing on here, you don't need to make a change.

I don't think it's the job of the caller to need to pass in the path to these proc files. The daemon (or whoever) should just be able to call `proc.GetNetworkData` and then the `proc` module would figure out the filename.

I again assume you did it this way for testing reasons, and it's totally fine that you did. But, I'll offer a couple of other ways to write this so as to allow for setting filename in tests while obscuring filenames from the caller.

### Option 1. Two functions.

caller
```go
proc.GetNetworkData()
```

proc
```go
func GetNetworkData() (stuff, err) {
	return getNetworkData(procNetDevPath)
}

func getNetworkData(path string) (stuff, err) {
	// do stuff
}
```

tests
```go
func TestGetNetworkData(t *testing.T) {
	out, err := getNetworkData("path/to/my/test/file")
	// etc
}
```

### Option 2. Overwrite global.

caller
```go
proc.GetNetworkData()
```

proc
```go
func GetNetworkData() (stuff, err) {
	path := procNetDevPath
	// do stuff
}
```

tests
```go
func TestGetNetworkData(t *testing.T) {
	origNetPath := procNetDevPath
	out, err := GetNetworkData()
	// etc

	procNetDevPath = origNetPath
}
```

-------

I kinda like option 1 best cuz there's no global business in the tests. Curious what your thoughts are. Maybe you already thought of these options? I'm curious to know 😃

Reply:
Ah nice thanks for the suggestions! I did have the path as a param for testing reasons, and I agree that the caller shouldn't be responsible for passing it in. I like option 1 better as well - it's clear and avoids global state. Updated to  do it that way